indico/modules/events/registration/util.py

```diff
@@ -995,3 +997,22 @@ def get_persons(registrations, include_accompanying_persons=False):
                             'registration': registration,
                             'is_accompanying': True} for person in registration.accompanying_persons)
     return persons
+
+
+def resize_uploaded_registration_picture(source, size):
+    """Resize uploaded registration picture to the given size."""
+    try:
+        picture = Image.open(source)
+        picture = ImageOps.exif_transpose(picture)
+        size_x, size_y = picture.size
+        if max(size_x, size_y) < size:
+            source.seek(0)
+            return source
+        ratio = float(size) / max(size_x, size_y)
+        picture = picture.resize((int(ratio * size_x), int(ratio * size_y)), Image.Resampling.BICUBIC)
+        image_bytes = BytesIO()
+        picture.save(image_bytes, 'PNG')
+        image_bytes.seek(0)
+        return image_bytes
+    except Exception:
+        return None
```

Comment:
Too broad IMHO, only catch exceptions you actually truly expect

Reply:
I changed it to except OSError

Reply:
What code in there is causing an OSError, and when?

Reply:
Opening or saving the image could throw an exception. But I was catching a broader exception so one doesn't slip by

Reply:
Did you ever encounter such an exception, e.g. by uploading a broken image? If not, I would rather not catch it but have it fail so there's a traceback etc in order to fix the problem or handle that specific exceptions later.

If you can cause an exception when opening a broken image, I'd only wrap the code where you're opening the image in the `try` block.

When saving I don't see why it would fail (you're not writing something to a folder on disk or anything like that)

Reply:
oh right, I didn't get an exception at that point yet, but I imagine our users would be able to make one very soon :)

Reply:
I left the try block for only when opening the image. While I cannot upload a fake image, I think we probably have some images that may be broken already and since we duplicate last registration personal data which includes the picture in our plugin, previously uploaded bad images could pass through